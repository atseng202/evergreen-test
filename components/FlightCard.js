/**
 * FlightCard Component
 * FlightCard contains parsed info of individual itineraries and legs
 * associated with the itinerary
 *
 * Props:
 * price, String like "Â£35"
 * agent, String like "Wizzair.com"
 * totalLegs, list of leg objects for this itinerary like,
 *  [{ id, departure_aiport, arrival_airport, departure_time,
 *    arrival_time, stops, airline_name, airline_id, duration_mins}, ...]
 *
 * State: None
 *
 * App -> FlightCard
 *
 * Use https://logos.skyscnr.com/images/airlines/favicon/{id}.png
 * for airline logos for each leg
 **/
function FlightCard({ price, agent, totalLegs }) {
  /**
   * Custom helper method to add leading 0 for hours and minutes
   * less than 0
   **/
  function formatZeros(time) {
    if (time < 10) {
      return `0${time}`;
    }
    return time;
  }

  /**
   * Custom helper method to parse and return the Leg component
   * for each list of legs associated with this itinerary
   **/
  function parseLegs() {
    return totalLegs.map((leg) => {
      const hours = Math.floor(leg.duration_mins / 60);
      const minutes = leg.duration_mins % 60;

      // Converting datetime string to presentational time
      const departureDate = new Date(leg.departure_time);
      const arrivalDate = new Date(leg.arrival_time);

      const formattedDepartureDate =
        formatZeros(departureDate.getHours()) +
        ":" +
        formatZeros(departureDate.getMinutes());

      const formattedArrivalDate =
        formatZeros(arrivalDate.getHours()) +
        ":" +
        formatZeros(arrivalDate.getMinutes());
      return (
        <div key={leg.id}>
          <img
            src={`https://logos.skyscnr.com/images/airlines/favicon/${leg.airline_id}.png`}
          ></img>
          <div>
            {formattedDepartureDate + "\n"}
            <br></br>
            {leg.departure_airport}
          </div>
          <br></br>
          <div>
            {formattedArrivalDate + "\n"}
            <br></br>
            {leg.arrival_airport}
          </div>
          <br></br>
          <div>
            {`${hours}h ${minutes}`}
            <br></br>
            {leg.stops > 1 ? leg.stops + " Stops" : "Direct"}
          </div>
          {/* {leg.airline_id} */}
        </div>
      );
    });
  }

  return (
    <div>
      {parseLegs()}
      <br></br>
      {price}
      <br></br>
      {agent}
      <br></br>
      <button>Select</button>
      <hr></hr>
    </div>
  );
}

export default FlightCard;
