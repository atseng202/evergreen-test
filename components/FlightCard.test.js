import React from "react";
import { render } from "@testing-library/react";
import FlightCard from "./FlightCard";

describe("FlightCard smoke and snapshot tests", () => {
  let expectedProps;

  beforeEach(() => {
    expectedProps = {
      price: "Â£35",
      agent: "Wizzair.com",
      totalLegs: [
        {
          id: "leg_1",
          departure_airport: "BUD",
          arrival_airport: "LTN",
          departure_time: "2020-10-31T15:35",
          arrival_time: "2020-10-31T17:00",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
        {
          id: "leg_4",
          departure_airport: "LTN",
          arrival_airport: "BUD",
          departure_time: "2020-11-11T19:45",
          arrival_time: "2020-11-11T21:10",
          stops: 0,
          airline_name: "Wizz Air",
          airline_id: "WZ",
          duration_mins: 145,
        },
      ],
    };
  });

  it("should render correctly", () => {
    render(<FlightCard {...expectedProps} />);
  });

  it("matches FlightCard snapshot", () => {
    const { asFragment } = render(<FlightCard {...expectedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
