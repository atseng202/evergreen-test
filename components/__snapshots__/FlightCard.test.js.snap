// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightCard smoke and snapshot tests matches FlightCard snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <img
        src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
      />
      <div>
        15:35

        <br />
        BUD
      </div>
      <br />
      <div>
        17:00

        <br />
        LTN
      </div>
      <br />
      <div>
        2h 25
        <br />
        Direct
      </div>
    </div>
    <div>
      <img
        src="https://logos.skyscnr.com/images/airlines/favicon/WZ.png"
      />
      <div>
        19:45

        <br />
        LTN
      </div>
      <br />
      <div>
        21:10

        <br />
        BUD
      </div>
      <br />
      <div>
        2h 25
        <br />
        Direct
      </div>
    </div>
    <br />
    Â£35
    <br />
    Wizzair.com
    <br />
    <button>
      Select
    </button>
    <hr />
  </div>
</DocumentFragment>
`;
